### ISSUE TITLE
Double clicking the button should toggle the tab between suspended and not

### ISSUE THREAD

**Issue by zwaa-vengeance** [deanoemcke/thegreatsuspender#179]
Double clicking the button should toggle the tab between suspended and not.

**Comment by deanoemcke** [comment]
@zwaa-vengeance which 'button' are you referring to?

**Comment by zwaa-vengeance** [comment]
That 'button' would be the Great Suspender icon next to the options icon at top right corner as seen below.
I just think that a Double Clicking option  would be great, it would save a lot of time using TGS.
Hope you agree. Sorry for late answer.
![thegreatsuspender](https://cloud.githubusercontent.com/assets/12023185/10246560/b70552ac-6912-11e5-807b-11080cb13f91.jpg)

**Comment by deanoemcke** [comment]
The upcoming release has a new keyboard shortcut to toggle the suspended state of the current tab. I may implement double click at some point in a future release.

**Comment by zwaa-vengeance** [comment]
I wish to emphasize that keyboard shortcuts is never a replacement for
mouse controls. Providing efficient mouse controls should be a priority
because keyboard shortcuts has distinct disadvantages:
- The more apps, extensions and whatnot that you use keyboard shortcuts in,
the more shortcuts you will inevitably confuse and forget. Using the wrong
shortcut can lead to devastating results, and therefore the *only* safe way
to operate is to use the mouse over keyboard shortcuts.
- There is a finite number of keyboard shortcuts that the operating system
provides, and many extensions that provide shortcuts doesn't allow for
changing them or turn them off, setting the table for unpredictable
conflicts even with shortcuts outside of Chrome, ie system wide shortcuts
that may be blocked. I have encountered this more than once and in a couple
of instances even chose to refrain from keeping the extension because of it.
- Over time, the average user will install and uninstall countless apps,
extensions etc, and even more keyboard shortcuts will accompany them.
Keeping track of all shortcuts for even one single app can be too
exhausting, and in the end the keyboard shortcuts that you remember and use
are the ones that you save the most time using, and that saved time is a
combination of how often you use them and how much faster any given
shortcut is compared to the mouse operation equivalent.
- For any developer to expect the users to memorize and use the keyboard
shortcuts and therefore not provide mouse alternatives is to deprive the
users their free choice of which shortcuts that they're going to try and
remember, if any at all (yes, such users exist and in abundance).

**Comment by deanoemcke** [comment]
I dont think that chrome gives me the ability to detect a double click on the extension icon (or any tab icons either).
Also, I would have thought that having to move the mouse all the way to the browser icon would be much harder than simply clicking on the large space in the middle of a suspended tab?

**Comment by fwextensions** [comment]
@deanoemcke it's possible for an extension to detect a double-click on the browser icon by checking the time between when the port between the popup and the background was opened vs. closed.  My [QuicKey](https://chrome.google.com/webstore/detail/quickey-%E2%80%93-the-quick-tab-s/ldlghkoiihaelfnggonhjnfiabmaficg) extension uses a double-click to toggle between the two most recent tabs.  (It's mostly triggered by double-pressing the <kbd>alt-Q</kbd> keyboard shortcut that opens the popup, but a double-click works the same.)

In the background script, in `chrome.runtime.onConnect` it adds a listener for `port.onDisconnect`: https://github.com/fwextensions/QuicKey/blob/master/src/js/background/background.js#L250

Then the first thing it does in the popup is call `chrome.runtime.connect();`:
https://github.com/fwextensions/QuicKey/blob/master/src/js/popup/init.js#L10

It tracks the time between the port connect and disconnect, and if it's < 450ms, it toggles to the previous tab.  In the TGS case, it would toggle the suspend state.  

It's not 100% reliable, especially on an old machine with Chrome running slowly, but it works well enough.  Listening to the port close is the only way to detect the popup getting closed, since the `beforeunload` event usually doesn't fire, but the port disconnect does.
